mlrun/feature_store/feature_set.py

```diff
@@ -479,8 +482,20 @@ def set_targets(
         else:
             self.spec.with_default_targets = False
 
-        validate_target_list(targets=targets)
+        self.spec.targets = []
+        self.add_targets(targets)
+
+        if default_final_step:
+            self.spec.graph.final_step = default_final_step
+
+    def add_targets(self, targets):
```

Comment:
the add_targets() is missing functionality of set_targets when used as an additional API (default_final_step, default_final_state).
let's just make it a helper internal function for now - name it __set_targets_add_targets_helper or similar